### PR TITLE
[Bugfix] Fix tsserver lsp setup. Fix ts-fmt-lint setup

### PR DESCRIPTION
**Pull request by PasiBergman**
# Description

1. Use nvim-lspconfig default root_dir configuration for tsserver
2. Fix ts-fmt-lint setup
   - prioritize local linter over global linter
   - fix `lintFormats` typo
   - only eslint_d supports `--fix-to-stdout`. Set formatter (i.e. "fix") only if eslint_d is used.

## How Has This Been Tested?

With a Typescript project using
- global eslint_d
- local eslint_d
- global eslint
- local eslint

### CODE REVIEW

File: lua/lsp/ts-fmt-lint.lua [hunk 1]
```diff
@@ -2,23 +2,41 @@
 -- You can look for project scope Prettier and Eslint with e.g. vim.fn.glob("node_modules/.bin/prettier") etc. If it is not found revert to global Prettier where needed.
 local M = {}
 
+vim.cmd "let proj = FindRootDirectory()"
```
Comment (abzcoding) [inline lua/lsp/ts-fmt-lint.lua:5]: can you move these lines to `M.setup` function  (L5 to L15)

Comment (PasiBergman) [reply lua/lsp/ts-fmt-lint.lua]: One moment. I made a incorrect git rebase...will fix soon.

Comment (PasiBergman) [reply lua/lsp/ts-fmt-lint.lua]: Lines have now been moved to `M.setup`. Note also, that I added `formattingSupported` to disable the formatting support on efm depending on `eslint_d`

**Comment by ChristianChiarulli** [comment]
I just noticed this seems to be broken,

**Pull request merged**